Feat: Fix img hash tag preview issue in corsawareimage.js issue #4015

jposeyjr opened the pull request:
A simple fix for https://github.com/4ian/GDevelop/issues/4015 by encoding the url and removing special chars as recommended by the MDN page. [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent) I also attached screen shots showing it now working and the break point showing full path with the # file name. 
![Screen Shot 2022-06-24 at 5 39 32 PM](https://user-images.githubusercontent.com/66893619/175720178-90dadf22-1fc4-42db-93fa-a198eecec920.png)
![Screen Shot 2022-06-24 at 5 39 54 PM](https://user-images.githubusercontent.com/66893619/175720186-6a4dd959-810b-484b-b64d-1ac650e47005.png)
![Screen Shot 2022-06-24 at 5 40 50 PM](https://user-images.githubusercontent.com/66893619/175720190-bbf91923-7998-407c-bae5-c5302b7398b2.png)

4ian commented:
Hi! Thanks for the PR.
Can you confirm images with a `#` will also work when previewing and exporting a game, both on the desktop app and the web-app?
Also, is there a reason for not using the encodeURIComponent function - could there be other "special" characters like `#` creating issues?

jposeyjr commented:
Yes it worked on all forms of previewing and exporting from what I could tell. If you have a list of all the different ones let me know and I can go through it for sure. I did start with encodeURIComponent at the start but it didn't fix the issue. I then added the regex to it as well and again it didn't fix it. So I limited the scope to simple the # and it was working, I didn't dig into why the encoded function wasn't working as I don't have a deep understanding of the code base.

jposeyjr commented:
@4ian I fixed the preview, that was something I thought was working but it wasn't that was my bad :). The export "worked" in that it exported them but there is an error in PIXI.JS when trying to view the game. It seems to encode the space but not the # so that is the same issue. Would that be an issue on PIXI.js repo ?? Or would you want the version we have modified? Sorry not my project so don't want to assume the workflow with vendor scripts. 

![Screen Shot 2022-06-26 at 12 09 45 PM](https://user-images.githubusercontent.com/66893619/175826456-197d196e-fa4f-440f-9a9f-17317fb3910e.png)
Exported project running index.html on live server extension.
![Screen Shot 2022-06-26 at 12 27 21 PM](https://user-images.githubusercontent.com/66893619/175826480-bba6c0de-2f2f-4a30-8e27-294e650ed13c.png)

jposeyjr commented:
Any updates on this?

jposeyjr commented:
So part of the problem is that img tags and the way urls work. That was what I fixed but hashtags shouldn’t be in urls at all for files or most things. As they are used for internal anchors in the url schematic. 

When I stepped through the breakpoints it seems pixi doesn’t encode urls either. Honestly all urls should be encoded regardless of solution just from best practices stand point and security.  

It doesn’t help how many different things touch and use the url with no good linking between them. Trying to find the best place to encode it was tough too, but that I leave to my inexperience with the code base.
